chainstate/src/detail/transaction_verifier/utils.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+
+use common::{
+    chain::{
+        tokens::{token_id, CoinOrTokenId, OutputValue, TokenData, TokensError},
+        Transaction,
+    },
+    primitives::Amount,
+};
+
+use super::error::ConnectTransactionError;
+
+pub fn insert_or_increase(
```

Comment:
I think this can be simplified with `total_amounts.entry(key).unwrap_or(Amount::from_atoms)`. I'm not sure abut the whole picture though.

Reply:
If it can return Err from closure, we might be use something like this: 
```rust
    total_amounts
        .entry(key)
        .and_modify(|value| {
            (*value + amount).ok_or(ConnectTransactionError::TokensError(
                TokensError::CoinOrTokenOverflow,
            ))?
        })
        .or_insert(amount);
```
Unfortunately, I haven't found a way to simplify it. If anyone can advise something, it will be great.